src/password_gen.rs

```diff
@@ -1,5 +1,40 @@
-// TODO: Draw the rest of the owl
-pub fn generate_password(seed: String) -> String {
-	println!("{:?}", seed);
-	String::from("TODO: some random pass")
+use rand::{rngs::OsRng, seq::SliceRandom, Rng};
+use sha2::{Digest, Sha256};
+
+use floem::reactive::use_context;
+
+use crate::env::Environment;
+
+pub fn generate_password(entropy: String) -> String {
+	let env = use_context::<Environment>().expect("No env context provider");
+
+	// Initialize RNG with system entropy
+	let mut rng = OsRng;
+
+	// Hash the byte array using SHA-256
+	let hash_result = Sha256::digest(entropy.as_bytes());
+
+	// Convert the resulting hash into a fixed-size byte array ([u8; 32])
+	let mut hash_array = [0u8; 32];
+	hash_array.copy_from_slice(&hash_result[..]);
+
+	// Mix additional entropy into the RNG
+	// We're doing nothing with the resulting array here
+	// We're just adding entropy to the system by doing stuff
+	// Not the greatest way to do this
+	rng.fill(&mut hash_array);
```

Comment:
Its very likely that the compiler will optimise this away entirely. Because `hash_array` isn't used, the compiler will probably remove the code. You could check the assembly to verify that.